x-pack/plugin/esql/compute/src/test/java/org/elasticsearch/compute/aggregation/ArrayStateTests.java

```diff
@@ -29,21 +38,37 @@ public static List<Object[]> params() {
         List<Object[]> params = new ArrayList<>();
 
         for (boolean inOrder : new boolean[] { true, false }) {
-            params.add(new Object[] { DataType.INTEGER, 1000, inOrder });
-            params.add(new Object[] { DataType.LONG, 1000, inOrder });
-            params.add(new Object[] { DataType.FLOAT, 1000, inOrder });
-            params.add(new Object[] { DataType.DOUBLE, 1000, inOrder });
-            params.add(new Object[] { DataType.IP, 1000, inOrder });
+            for (IntSupplier count : new IntSupplier[] { new Fixed(100), new Fixed(1000), new Random(100, 5000) }) {
+                params.add(new Object[] { DataType.INTEGER, count, inOrder });
+                params.add(new Object[] { DataType.LONG, count, inOrder });
+                params.add(new Object[] { DataType.FLOAT, count, inOrder });
+                params.add(new Object[] { DataType.DOUBLE, count, inOrder });
+                params.add(new Object[] { DataType.IP, count, inOrder });
+            }
```

Comment:
More lengths makes it more likely we'll expose this bug. Including the randomized one.